### DISCUSSION TITLE
What will happen if pulsar-proxy crashes.

### DISCUSSION THREAD

**Discussion by umialpha** [apache/pulsar#18790] [Q&A] [answered]
Hi, I want some clarification about the crash of pulsar-proxy.

Will the messages that unacknowledged  be redelivered immediately?( e.g. shared subscription).

If the answer is yes. I think it needs to be clarified in the official doc because users may be get confused when different consumers get the same message without any client's disconnection.

**Comment by jiazhai** [comment, marked as answer]
hi @umialpha The behavior "when different consumers get the same message without any client's disconnection" is not related with proxy.  it seems the default behavior for shared-sub. the unacked message get re-delivered once message-ack-timeout, and the target consumer may not be the same consumer that handled this message before.
